python/TestHarness/TestHarness.py

```diff
@@ -357,16 +357,32 @@ def findAndRunTests(self, find_only=False):
         self.preRun()
         self.start_time = clock()
         launched_tests = []
+
+        #used for -i
         if self.options.input_file_name != '':
+
             self._infiles = self.options.input_file_name.split(',')
 
-        if self.options.spec_file and os.path.isdir(self.options.spec_file):
-            search_dir = self.options.spec_file
-        elif self.options.spec_file and os.path.isfile(self.options.spec_file):
-            search_dir = os.path.dirname(self.options.spec_file)
-            self._infiles = [os.path.basename(self.options.spec_file)]
+            if(os.path.dirname(self._infiles[0]) !=''):
```

Comment:
Oops! This is a critical error here! Indenting in Python is significant. This `if` statement is nested underneath the one above, so you'd only be testing this logic when `-i` is supplied, which is logically incorrect.